pkg/kubeconfig/kubeconfig.go

```diff
@@ -126,17 +148,22 @@ func (cfg *configLoader) GetConfig() error {
 		if err != nil {
 			return err
 		}
+	} else {
+		existingKubeConfig, err = clientcmd.LoadFromFile(localPath)
+		if err != nil {
+			return err
+		}
 	}
 
-	f, err := os.OpenFile(localPath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	mergedKubeConfigBytes, err := clientcmd.Write(*mergeKubeconfigs([]*clientcmdapi.Config{existingKubeConfig, newKubeConfig}))
```

Comment:
We could simplify this with [`clientcmd.WriteToFile`](https://godoc.org/k8s.io/client-go/tools/clientcmd#WriteToFile).

Reply:
Good catch. Thanks :)